### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW05)

### PR DESCRIPTION
**Pull request by KornevNikita**
LLVM: llvm/llvm-project@897e345
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@77ee97a

### CODE REVIEW

**Comment by KornevNikita** [comment]
Working on the reset of failing tests.

File: llvm/test/DebugInfo/Generic/assignment-tracking/instcombine/alloca-bitcast.ll [hunk 1]
```diff
@@ -1,4 +1,4 @@
-; RUN: opt -passes=instcombine -S %s -o - -experimental-assignment-tracking \
+; RUN: opt -passes=instcombine -S %s -o - \
```
Comment (bader) [inline llvm/test/DebugInfo/Generic/assignment-tracking/instcombine/alloca-bitcast.ll:1]: This doesn't seem to be related to DPC++ compiler. Why do we need this change?
Could you add a comment to the code?

File: llvm/unittests/Analysis/MLModelRunnerTest.cpp [hunk 2]
```diff
@@ -130,7 +130,7 @@ TEST(ReleaseModeRunner, ExtraFeaturesOutOfOrder) {
 }
 
 #if defined(LLVM_ON_UNIX)
-TEST(InteractiveModelRunner, Evaluation) {
+TEST(InteractiveModelRunner, DISABLED_Evaluation) {
```
Comment (bader) [inline llvm/unittests/Analysis/MLModelRunnerTest.cpp:133]: What's wrong here?

File: llvm/test/Transforms/InstCombine/icmp-binop.ll [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
-; RUN: opt -opaque-pointers < %s -passes=instcombine -S | FileCheck %s
+; RUN: opt -opaque-pointers -opaque-pointers < %s -passes=instcombine -S | FileCheck %s
```
Comment (bader) [inline llvm/test/Transforms/InstCombine/icmp-binop.ll:2]: No need to make this change.

Comment (bader) [inline llvm/test/Transforms/InstCombine/icmp-binop.ll:2]: No need to make this change here.

File: llvm/test/DebugInfo/Generic/assignment-tracking/instcombine/alloca-bitcast.ll (hunk 1 above)
Comment (KornevNikita) [reply llvm/test/DebugInfo/Generic/assignment-tracking/instcombine/alloca-bitcast.ll:1]: I'm assuming there was a wrong conflict resolution, so I just brought it up to date with the llorg version.

File: llvm/unittests/Analysis/MLModelRunnerTest.cpp (hunk 2 above)
Comment (KornevNikita) [reply llvm/unittests/Analysis/MLModelRunnerTest.cpp:133]: It hangs. Turned off, so it doesn't affect testing. Looking for a fix.

Comment (bader) [reply llvm/unittests/Analysis/MLModelRunnerTest.cpp:133]: Okay. I hope you manage to fix it before the merge. If not, please, file a bug report to track the status of the issue.

File: llvm/test/Transforms/InstCombine/icmp-binop.ll (hunk 3 above)
Comment (KornevNikita) [reply llvm/test/Transforms/InstCombine/icmp-binop.ll]: Thanks, removed c20bf86

**Comment by KornevNikita** [comment]
@bader I guess we're ready to merge. All failures seem to be unrelated. Take a look please

**Comment by bader** [comment]
```
[334/335] Generating html Sphinx documentation for clang into "/home/runner/work/llvm/llvm/build/tools/clang/docs/html"
FAILED: tools/clang/docs/CMakeFiles/docs-clang-html /home/runner/work/llvm/llvm/build/tools/clang/docs/CMakeFiles/docs-clang-html 
cd /home/runner/work/llvm/llvm/build/tools/clang/docs && /usr/local/bin/cmake -E env /usr/local/bin/sphinx-build -b html -d /home/runner/work/llvm/llvm/build/tools/clang/docs/_doctrees-clang-html -q -t builder-html -D version=17 -D release=17.0.0git -tPreRelease -W /home/runner/work/llvm/llvm/build/tools/clang/docs /home/runner/work/llvm/llvm/build/tools/clang/docs/html
/usr/local/lib/python3.10/dist-packages/recommonmark/parser.py:75: UserWarning: Container node skipped: type=document
  warn("Container node skipped: type={0}".format(mdnode.t))

Warning, treated as error:
/home/runner/work/llvm/llvm/build/tools/clang/docs/ClangNvlinkWrapper.rst:document isn't included in any toctree
ninja: build stopped: subcommand failed.
Error: Process completed with exit code 1.

```
I see that `Generate Doxygen documentation` passes in the latest nightly, so this error is not expected. Is this a bug we pulled from the community code? Is there a fix for this we can pull as well?

**Comment by bader** [comment]
FAIL: SYCL :: GroupAlgorithm/barrier.cpp (670 of 1231)

https://github.com/intel/llvm/actions/runs/4164875887/jobs/7207988657 

Is this expected? I haven't seen this error before.

**Comment by bader** [comment]
Other than the two issues in my previous comments. Everything else looks good to me.

**Comment by KornevNikita** [comment]
> ```
> [334/335] Generating html Sphinx documentation for clang into "/home/runner/work/llvm/llvm/build/tools/clang/docs/html"
> FAILED: tools/clang/docs/CMakeFiles/docs-clang-html /home/runner/work/llvm/llvm/build/tools/clang/docs/CMakeFiles/docs-clang-html 
> cd /home/runner/work/llvm/llvm/build/tools/clang/docs && /usr/local/bin/cmake -E env /usr/local/bin/sphinx-build -b html -d /home/runner/work/llvm/llvm/build/tools/clang/docs/_doctrees-clang-html -q -t builder-html -D version=17 -D release=17.0.0git -tPreRelease -W /home/runner/work/llvm/llvm/build/tools/clang/docs /home/runner/work/llvm/llvm/build/tools/clang/docs/html
> /usr/local/lib/python3.10/dist-packages/recommonmark/parser.py:75: UserWarning: Container node skipped: type=document
>   warn("Container node skipped: type={0}".format(mdnode.t))
> 
> Warning, treated as error:
> /home/runner/work/llvm/llvm/build/tools/clang/docs/ClangNvlinkWrapper.rst:document isn't included in any toctree
> ninja: build stopped: subcommand failed.
> Error: Process completed with exit code 1.
> ```
> 
> I see that `Generate Doxygen documentation` passes in the latest nightly, so this error is not expected. Is this a bug we pulled from the community code? Is there a fix for this we can pull as well?

Strange thing. https://github.com/llvm/llvm-project/commit/6f96dadb7a5b1593a0b6348539f66f7e945f2561: "git branch --contains 6f96dadb7a5b1593a0b6348539f66f7e945f2561" said that the branch has this commit, but at the same time I have deleted by this commit .rst files in my repo. Cherry-picked it with https://github.com/intel/llvm/pull/8184/commits/72d11625f4f88dfd86de873e43c4fe76bc006e3d, should pass now.

**Comment by KornevNikita** [comment]
> FAIL: SYCL :: GroupAlgorithm/barrier.cpp (670 of 1231)
> 
> https://github.com/intel/llvm/actions/runs/4164875887/jobs/7207988657
> 
> Is this expected? I haven't seen this error before.

Thanks for mentioning, no, need to investigate.

**Comment by KornevNikita** [comment]
So, looks like 

> > ```
> > [334/335] Generating html Sphinx documentation for clang into "/home/runner/work/llvm/llvm/build/tools/clang/docs/html"
> > FAILED: tools/clang/docs/CMakeFiles/docs-clang-html /home/runner/work/llvm/llvm/build/tools/clang/docs/CMakeFiles/docs-clang-html 
> > cd /home/runner/work/llvm/llvm/build/tools/clang/docs && /usr/local/bin/cmake -E env /usr/local/bin/sphinx-build -b html -d /home/runner/work/llvm/llvm/build/tools/clang/docs/_doctrees-clang-html -q -t builder-html -D version=17 -D release=17.0.0git -tPreRelease -W /home/runner/work/llvm/llvm/build/tools/clang/docs /home/runner/work/llvm/llvm/build/tools/clang/docs/html
> > /usr/local/lib/python3.10/dist-packages/recommonmark/parser.py:75: UserWarning: Container node skipped: type=document
> >   warn("Container node skipped: type={0}".format(mdnode.t))
> > 
> > Warning, treated as error:
> > /home/runner/work/llvm/llvm/build/tools/clang/docs/ClangNvlinkWrapper.rst:document isn't included in any toctree
> > ninja: build stopped: subcommand failed.
> > Error: Process completed with exit code 1.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I see that `Generate Doxygen documentation` passes in the latest nightly, so this error is not expected. Is this a bug we pulled from the community code? Is there a fix for this we can pull as well?
> 
> Strange thing. [llvm/llvm-project@6f96dad](https://github.com/llvm/llvm-project/commit/6f96dadb7a5b1593a0b6348539f66f7e945f2561): "git branch --contains [6f96dad](https://github.com/intel/llvm/commit/6f96dadb7a5b1593a0b6348539f66f7e945f2561)" said that the branch has this commit, but at the same time I have deleted by this commit .rst files in my repo. Cherry-picked it with [72d1162](https://github.com/intel/llvm/commit/72d11625f4f88dfd86de873e43c4fe76bc006e3d), should pass now.

Oops, looks like we're still using this the function deleted by this commit. Need to investigate to find the right solution.

**Comment by KornevNikita** [comment]
@bader I'm having troubles trying reproduce this failure on sycl-web to figure out if it is already fixed by community. Can we merge it as is and temporarily disable the test to see if it pass with the next pulldown?

**Comment by KornevNikita** [comment]
> > @bader I'm having troubles trying to reproduce this failure on sycl-web to figure out if it is already fixed by community. Can we merge it as is and temporarily disable the test to see if it passes with the next pulldown?
> 
> What is "this failure"? Why don't you just merge the version of sycl-web, which passes the test?

"this failure" is FAIL: SYCL :: GroupAlgorithm/barrier.cpp (670 of 1231). We don't see it now as the workflow was queued for a few days and probably canceled.
I don't know if the last sycl-web passes this test as I'm having troubles compiling it on available for me machines with cuda support. I can try to merge the last sycl-web state here, but it will lead to a new lit failures because it's unstable.

**Comment by bader** [comment]
> > > @bader I'm having troubles trying to reproduce this failure on sycl-web to figure out if it is already fixed by community. Can we merge it as is and temporarily disable the test to see if it passes with the next pulldown?
> > 
> > 
> > What is "this failure"? Why don't you just merge the version of sycl-web, which passes the test?
> 
> "this failure" is FAIL: SYCL :: GroupAlgorithm/barrier.cpp (670 of 1231). We don't see it now as the workflow was queued for a few days and probably canceled. I don't know if the last sycl-web passes this test as I'm having troubles compiling it on available for me machines with cuda support. I can try to merge the last sycl-web state here, but it will lead to a new lit failures because it's unstable.

Please, merge tip of the `sycl` branch to your branch to get latest results. I hope CI system is more stable now and will be able to show the results. If they are clean, we should be good to go.

**Comment by KornevNikita** [comment]
I'm fixing new failures.

**Comment by KornevNikita** [comment]
Closing this, because https://github.com/intel/llvm/pull/8412 happened.
